src/V3Config.cpp

```diff
@@ -145,9 +146,310 @@ class V3ConfigIgnores {
+    typedef std::map<string,VarAttrMap> VarAttrMapFTask; //< Map from ftask (wcard) to map of vars
+    typedef std::map<string,VarAttrMapFTask> AttrMap; //< Map from module (wcard) to map of ftask and below
+    typedef std::unordered_set<std::string> FunctionList; //< List of functions for isolation
+
+    AttrMap m_varAttrs; //< Configured attributes
+    std::map<std::string,FunctionList> m_functionIsolate; //< Configured function isolations
+
+    static V3ConfigVarAttr s_singleton;  // Singleton (not via local static, as that's slow)
+    V3ConfigVarAttr() {}
+    ~V3ConfigVarAttr() {}
+
+    void applyVarAttr(AstVar* varp, AttrEntry entry) {
+        AstNode* newp = new AstAttrOf(varp->fileline(), entry.type);
+        varp->addAttrsp(newp);
+        if (entry.type == AstAttrType::VAR_PUBLIC_FLAT_RW) {
+            newp->addNext(new AstAlwaysPublic(varp->fileline(), entry.sentreep, NULL));
+        }
+    }
+    void applyVarAttrForTask(const VarAttrMap& map, AstVar* varp) {
+        const string var = varp->name();
+        for (VarAttrMap::const_iterator it = map.begin(); it != map.end(); ++it) {
+            if (VString::wildmatch(var.c_str(), it->first.c_str())) {
+                for (AttrEntryVector::const_iterator eit = it->second.begin();
+                     eit != it->second.end(); ++eit) {
+                    applyVarAttr(varp, *eit);
+                }
+            }
+        }
+    }
+    void applyVarAttrForModule(const VarAttrMapFTask& map, const string& ftask, AstVar* varp) {
+        for (VarAttrMapFTask::const_iterator it = map.begin(); it != map.end(); ++it) {
+            if (VString::wildmatch(ftask.c_str(), it->first.c_str())) {
+                applyVarAttrForTask(it->second, varp);
+            }
+        }
+    }
+    void applyFunctionIsolate(const FunctionList& list, AstNodeFTask* funcp) {
+        const string& func = funcp->name();
+        for (FunctionList::const_iterator it = list.begin(); it != list.end(); ++it) {
+            if (VString::wildmatch(func.c_str(), it->c_str())) {
+                funcp->attrIsolateAssign(true);
+            }
+        }
+    }
+public:
+    inline static V3ConfigVarAttr& singleton() { return s_singleton; }
+
+    void addVarAttr(FileLine* fl, const string& module, const string& ftask, const string& var,
+                    AstAttrType type, AstSenTree* sensep) {
+        m_varAttrs[module][ftask][var].push_back({type, sensep});
+    }
+    void addFunctionIsolate(const string& module, const string& function) {
+        m_functionIsolate[module].insert(function);
+    }
+
+    void applyVarAttr(const AstNodeModule* modulep, const AstNodeFTask* ftaskp, AstVar* varp) {
+        const string& module = modulep->name();
+        const string& ftask = (ftaskp ? ftaskp->name() :  "");
+
+        for (AttrMap::const_iterator it = m_varAttrs.begin(); it != m_varAttrs.end(); ++it) {
+            if (VString::wildmatch(module.c_str(), it->first.c_str())) {
```

Comment:
If wildmatch doesn't have a flavor that takes two std strings please add one, then remove c_str()s here.